usr/src/test/elf-tests/tests/sysroot/test-search.sh

```diff
@@ -0,0 +1,94 @@
+#! /bin/ksh
+#
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+# Copyright 2023 Richard Lowe
+
+#
+# Test that searching for sysroot paths works as we expect.
+#
+
+set -o pipefail
+
+tmpdir=$(mktemp -td test-search.XXXXXX)
+if [[ -z $tmpdir ]]; then
+	print -u2 "Failed to create temporary directory"
+	exit 1
+fi
+
+cd $tmpdir
+
+cleanup() {
+	cd /
+	rm -fr $tmpdir
+}
+
+trap 'cleanup' EXIT
+
+misc_fail() {
+	print -u2 "$@"
+	exit 1
+}
+
+if [[ $PWD != $tmpdir ]]; then
+	misc_fail "Failed to create temporary directory: $tmpdir"
```

Comment:
This comment doesn't feel right. At this point it's now about no longer being able to cd, no?